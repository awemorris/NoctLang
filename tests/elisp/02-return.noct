func main() {
    var a = 0;
    if (a == 0) {
        return 123;
    }
    a = 1;
    return 321;
}
