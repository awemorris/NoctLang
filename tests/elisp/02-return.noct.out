(defun main ()
  (let ((a 0))
    (catch 'early-return
      (setq a 0)
      (if (= a 0)
        (progn
          (throw 'early-return 123)
        )
      )
      (setq a 1)
      (throw 'early-return 321)
      0
    )
  )
)
